From 88048c200486d7bf7fc81196798dbeefde6b80eb Mon Sep 17 00:00:00 2001
From: Lang Hames <lhames@gmail.com>
Date: Sat, 18 Aug 2018 18:38:37 +0000
Subject: [PATCH] [RuntimeDyld] Fix a bug in RuntimeDyld::loadObjectImpl that
 was over-allocating space for common symbols.

Patch by Dmitry Sidorov. Thanks Dmitry!

Differential revision: https://reviews.llvm.org/D50240

git-svn-id: https://llvm.org/svn/llvm-project/llvm/trunk@340125 91177308-0d34-0410-b5e6-96231b3b80d8
---
 lib/ExecutionEngine/RuntimeDyld/RuntimeDyld.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/ExecutionEngine/RuntimeDyld/RuntimeDyld.cpp b/lib/ExecutionEngine/RuntimeDyld/RuntimeDyld.cpp
index af5427e..df4bd572 100644
--- a/lib/ExecutionEngine/RuntimeDyld/RuntimeDyld.cpp
+++ b/lib/ExecutionEngine/RuntimeDyld/RuntimeDyld.cpp
@@ -277,7 +277,7 @@ RuntimeDyldImpl::loadObjectImpl(const object::ObjectFile &Obj) {
           uint64_t Size = I->getCommonSize();
           if (!CommonAlign)
             CommonAlign = Align;
-          CommonSize += alignTo(CommonSize, Align) + Size;
+          CommonSize = alignTo(CommonSize, Align) + Size;
           CommonSymbolsToAllocate.push_back(*I);
         }
       } else
-- 
1.8.3.1

